### PR TITLE
Updating ORA2 documentation

### PR DESCRIPTION
**Pull request by Andytr1**
**TL;DR -** Conform to [OEP-19](https://github.com/edx/open-edx-proposals/blob/master/oeps/oep-0019-bp-developer-documentation.rst) by moving and updating documentation to be close to code.

**What changed?**

- [ More in depth breakdown of changes ]
- [ Peripheral things that got changed ]
- [ etc... ]

**Developer Checklist**
- [ ] Reviewed the [release process](./release_process.md)
- [ ] Translations up to date
- [ ] JS minified, SASS compiled
- [ ] Test suites passing on Jenkins
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5241](https://openedx.atlassian.net/browse/EDUCATOR-5241)

@edx/masters-devs-gta

### CODE REVIEW

File: docs/developers_guide.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+- ``cd /edx/src/edx-ora2`` # goto into ora2 folder
+
+- ``source edx-ora2/bin/activate`` # activate ora2 virtual environment
+
+- ``pip install -U edx-submissions`` # installs latest version of submissions from pypi
+
+          OR
+          
+- ``pip install -e /edx/src/edx-submissions`` # istalls local version of submissions
```
Comment (nsprenkle) [inline docs/developers_guide.rst:60]: Just a note that this renders strangely in GitHub's preview. Consider reformatting.

File: docs/ora_team_submissions.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+There’s an AssessmentWorkflow model that has a workflow app label, which means that the table name is prefixed with `workflow_` and not `assessment_` like everything else (I’m sure it’s to help resolve a name collision).  `See for reference: <https://github.com/edx/edx-ora2/blob/001a6804e605426fdc43d4b5d83a38051c756c8d/openassessment/workflow/models.py#L108>`_.  In fact, each of the AssessmentWorkflow, AssessmentWorkflowStep, and AssessmentWorkflowCancellation models live under the workflow app label.
+
+- The ``SubmissionMixin.create_submission()`` method calls the ``workflow_api.create_workflow()`` method  (it stops at ``WorkflowMixin.create_workflow()`` along the way).  This is the entry point to workflow management in ORA.
+
+- ``create_workflow()`` calls ``AssessmentWorkflow.start_workflow()`` 
+
+ - …which calls the ``on_init()`` method of each specific type of step (e.g. self, peer, staff).
+
+ - The ``on_init()`` method of e.g. StaffWorkflow is what creates a StaffWorkflow record associated with a given course, item, and submission (and eventually, a scorer and an assessment).
+
```
Comment (nsprenkle) [inline docs/ora_team_submissions.rst:18]: Formatting for these shows as a block quote w/ bulleted list instead of correctly indenting.

File: docs/ora_team_submissions.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+Staff Workflows
+---------------
+
+These should be the only workflow type that we care about for team submissions.
+
+`Python API for managing StaffWorkflow data: <https://github.com/edx/edx-ora2/blob/master/openassessment/assessment/api/staff.py>`_:
+
+- **on_init(submission_uuid)**
+
+ - Looks up submission record from Submissions API.  Then creates a StaffWorkflow associated with it.
```
Comment (nsprenkle) [inline docs/ora_team_submissions.rst:29]: Formatting for this shows as a block quote w/ bulleted list instead of correctly indenting.

File: docs/ora_team_submissions.rst [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+**For teams:**
+
+- We will have one workflow per team - there’s no value in creating e.g. 5 workflows for a team of 5 learners when we only want 1 of those 5 workflows to be acted on by course staff.
+
+- From the perspective of grading, it doesn’t matter, as long as we end up in a state where there is one Score per learner.
+
+Data Models Design
+------------------
+
```
Comment (nsprenkle) [inline docs/ora_team_submissions.rst:39]: Redundant line

File: docs/ora_team_submissions.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+- Create a new ``TeamStaffWorkflow`` model that inherits from ``StaffWorkflow``.
+
+- Create a new ``TeamSubmission`` model with a many to many relationship to ``Submission``.  The ``TeamStaffWorkflow`` now references a ``TeamSubmission``.
+
+- The TeamSubmission model also records who submitted on behalf of the team.
+
+- We’ll actually also need a new, team-centric version of the ``workflow.AssessmentWorkflow`` model, so that we can pass a ``team_submission_uuid`` to the ``start_workflow()`` method.  It will follow the same flow of logic of creating a workflow record and associated steps.
+
+ - ``AssessmentWorkflow.start_workflow()`` calls the ``on_init()`` function of each `assessment type’s API, e.g. <https://github.com/edx/edx-ora2/blob/master/openassessment/assessment/api/staff.py#L59>`_ In fact, pretty much all of the functions provided by assessment API’s require a ``submission_uuid`` parameter.  So we’ll likely want some new API like ``assessment.api.team`` that provides the same functionality, but takes a ``team_submission_uuid`` as the first parameter.
```
Comment (nsprenkle) [inline docs/ora_team_submissions.rst:60]: Formatting for this shows as a block quote w/ bulleted list instead of correctly indenting.

File: docs/developers_guide.rst [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+-----------------
+ORA is broken into two separate repositories:
+
+- `edx-ora2 <https://github.com/edx/edx-ora2>`_
+
+- `edx-submissions <https://github.com/edx/edx-submissions>`_
+
+Using ORA with docker devstack
+------------------------------
+Note - "from inside the lms", means you've run 'make lms-shell' from the devstack directory and are on a command prompt inside the lms container.
```
Comment (nsprenkle) [inline docs/developers_guide.rst:16]: Make this render as code.

```suggestion
Note - "from inside the lms", means you've run ``make lms-shell`` from the devstack directory and are on a command prompt inside the lms container.
```

File: docs/developers_guide.rst [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+1. Clone edx-ora2 repo into ../src/ directory (relative to your 'devstack' repo location). 
+
+2. From the src directory, run the following command to install ora2 in the lms's and studio's virtual environments:
+
+ ``make install-local-ora``
```
Comment (nsprenkle) [inline docs/developers_guide.rst:22]: This is improperly indented to show as a code block

File: docs/developers_guide.rst [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+Alternatively, one can log into either lms (``make lms-shell``) or studio (``make studio-shell``) and execute the following commands to uninstall ora2 and reinstall your local copy;
+ 
+ ``pip uninstall ora2 -y; pip install -e /edx/src/edx-ora2/``
```
Comment (nsprenkle) [inline docs/developers_guide.rst:26]: This is improperly indented to show as a code block

File: docs/release_process.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+- create a release `tag on GitHub <https://github.com/edx/edx-ora2/releases>`_
+- ORA2 is a dependency in platform. If immediate release is needed, create PR to update `requirements files` in `edx-platform`. See `PR for reference <https://github.com/edx/edx-platform/pull/24830>`_ . Otherwise, a bot will automatically update the requirements in platform.
+- If manual testing of the changes against edx-platform is desired, create a sandbox.
+- get green build on the edx-platform PR, merge
+- manually test changes on stage
+
+Updating translations
+---------------------
+
+If you change any text that is ultimately wrapped by ugettext (in a python, JS, or HTML file), you'll have to update translations.
```
Comment (nsprenkle) [inline docs/release_process.rst:21]: This is a better set of instructions for what is in the developer's guide. Suggest we have only one and link to the source-of-truth.

**Review by nsprenkle — COMMENTED** [review]
Documentation looks good, but a few formatting pieces copied over strangely. Specifically: bulleted lists that are indented incorrectly show as block quotes in the "Rich Diff" view.

Want to correct those, but otherwise GTG.

File: docs/release_process.rst (hunk 9 above)
Comment (Andytr1) [reply docs/release_process.rst:21]: Counter argument for keeping them in dev guide: they are listed as discreet commands for people to reference, while this guide is specifically around release processes.

**Review by nsprenkle — APPROVED** [review]
Beautiful :shipit:

**Review by jansenk — APPROVED** [review]
lgtm
